src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -1123,132 +1123,80 @@ int sd_dhcp_server_set_default_lease_time(sd_dhcp_server *server, uint32_t t) {
         return 1;
 }
 
-int sd_dhcp_server_set_dns(sd_dhcp_server *server, const struct in_addr dns[], unsigned n) {
-        assert_return(server, -EINVAL);
-        assert_return(dns || n <= 0, -EINVAL);
-
-        if (server->n_dns == n &&
-            memcmp(server->dns, dns, sizeof(struct in_addr) * n) == 0)
-                return 0;
-
-        if (n <= 0) {
-                server->dns = mfree(server->dns);
-                server->n_dns = 0;
-        } else {
-                struct in_addr *c;
-
-                c = newdup(struct in_addr, dns, n);
-                if (!c)
-                        return -ENOMEM;
+int sd_dhcp_server_set_servers(sd_dhcp_server *server,
+                               sd_dhcp_lease_info what,
+                               const struct in_addr addresses[],
+                               unsigned n_addresses) {
```

Comment:
if we fix that, maybe also change the last argument to size_t? it's memory, hence should be measured in size_t

Reply:
hmm? it specifies the number of structues in_addr. i.e. that's plain memory. i.e. its the reason why calloc() and reallocarray() take a pair of size_t arguments... for a ll new code we measure the size of arrays in size_t these days...

Reply:
This can be done in a separate PR. It's not really related to this one.